integration_tests/src/testcases/locator.test.ts

```diff
@@ -0,0 +1,415 @@
+        lsm_redeem_threshold: 2,
+        lsm_min_bond_amount: '1000',
+        lsm_redeem_max_interval: 60_000,
+        bond_limit: '100000',
+        min_stake_amount: '2',
+        icq_update_delay: 5,
+      },
+      staker_params: {
+        min_stake_amount: '10000',
+        min_ibc_transfer: '10000',
+      },
+    };
+    const factory2_instantiate_message = factory1_instantiate_message;
+    factory2_instantiate_message.salt = '2';
+    const instantiate1Res = await DropFactory.Client.instantiate(
+      client,
+      account.address,
+      codeIds.factory,
+      factory1_instantiate_message,
+      'drop-staking-factory',
+      'auto',
+      [],
+    );
+    const instantiate2Res = await DropFactory.Client.instantiate(
+      client,
+      account.address,
+      codeIds.factory,
+      factory2_instantiate_message,
+      'drop-staking-factory',
+      'auto',
+      [],
+    );
+    expect(instantiate1Res.contractAddress).toHaveLength(66);
+    expect(instantiate2Res.contractAddress).toHaveLength(66);
+    expect(instantiate1Res.contractAddress).not.toBe(
+      instantiate2Res.contractAddress,
+    );
+    context.contracts.factories = [
+      new DropFactory.Client(client, instantiate1Res.contractAddress),
+      new DropFactory.Client(client, instantiate2Res.contractAddress),
+    ];
+  });
+  it('Instantiate locator contract', async () => {
+    const { client, account, codeIds } = context;
+    const instantiateRes = await DropLocator.Client.instantiate(
+      client,
+      account.address,
+      codeIds.locator,
+      {},
+      'drop-locator',
+      'auto',
+      [],
+    );
+    expect(instantiateRes.contractAddress).toHaveLength(66);
+    context.contracts.locator = new DropLocator.Client(
+      client,
+      instantiateRes.contractAddress,
+    );
+  });
+  it('Add factory instances to locator contract & Make sure they were added', async () => {
+    const { locator, factories } = context.contracts;
+    const { account } = context;
+    await locator.addChains(
+      account.address,
+      {
+        chains: [
+          {
+            name: 'instance_1',
+            factory_addr: factories[0].contractAddress,
+          },
+          {
+            name: 'instance_2',
+            factory_addr: factories[1].contractAddress,
+          },
+        ],
+      },
+      'auto',
+      '',
+      [],
+    );
+    expect((await locator.queryChains()).length).toEqual(2);
+    expect((await locator.queryFactoryInstances()).length).toEqual(2);
+  });
+  it('Query factory instances', async () => {
+    const { locator } = context.contracts;
+    const res = await locator.queryFactoryInstances();
+    expect(res).toHaveLength(2);
+    let matches: number = 0;
+    for (const factory of context.contracts.factories) {
+      const factoryState = await factory.queryState();
+      for (const instance of res) {
+        if (instance.addr === factory.contractAddress) {
+          matches += 1;
```

Comment:
no need, you can compare length of `locator.queryFactoryInstances()` and `context.contracts.factories`